File: gachisa-backend/src/main/java/com/gachisa/payment/dto/RefundResponse.java
package com.gachisa.payment.dto;

import com.gachisa.payment.entity.Refund;
import com.gachisa.payment.entity.RefundStatus;
import java.time.LocalDateTime;

public record RefundResponse(
        Long refundId,
        Long paymentId,
        int amount,
        String reason,
        RefundStatus status,
        int retryCount,
        LocalDateTime nextRetryAt,
        String failureCode,
        String failureMessage,
        LocalDateTime requestedAt,
        LocalDateTime refundedAt
) {

    public static RefundResponse from(Refund refund) {
        return new RefundResponse(
                refund.getId(),
                refund.getPaymentId(),
                refund.getAmount(),
                refund.getReason(),
                refund.getStatus(),
                refund.getRetryCount(),
                refund.getNextRetryAt(),
                refund.getFailureCode(),
                refund.getFailureMessage(),
                refund.getRequestedAt(),
                refund.getRefundedAt()
        );
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/payment/dto/RefundResponse.kt
package com.gachisa.payment.dto

import com.gachisa.payment.entity.Refund
import com.gachisa.payment.entity.RefundStatus
import java.time.LocalDateTime

data class RefundResponse(
    val refundId: Long,
    val paymentId: Long,
    val amount: Int,
    val reason: String,
    val status: RefundStatus,
    val retryCount: Int,
    val nextRetryAt: LocalDateTime?,
    val failureCode: String?,
    val failureMessage: String?,
    val requestedAt: LocalDateTime,
    val refundedAt: LocalDateTime?,
) {
    companion object {
        fun from(refund: Refund): RefundResponse = RefundResponse(
            refundId = refund.id!!,
            paymentId = refund.paymentId,
            amount = refund.amount,
            reason = refund.reason,
            status = refund.status,
            retryCount = refund.retryCount,
            nextRetryAt = refund.nextRetryAt,
            failureCode = refund.failureCode,
            failureMessage = refund.failureMessage,
            requestedAt = refund.requestedAt,
            refundedAt = refund.refundedAt,
        )
    }
}
